sys/kern/interrupt.c

```diff
@@ -60,14 +64,26 @@ static void insert_handler(intr_event_t *ie, intr_handler_t *ih) {
   ie->ie_count++;
 }
 
-void intr_event_add_handler(intr_event_t *ie, intr_handler_t *ih) {
+static void ie_add_handler(intr_event_t *ie, intr_handler_t *ih) {
   WITH_SPIN_LOCK (&ie->ie_lock) {
-    insert_handler(ie, ih);
+    ie_insert_handler(ie, ih);
     if (ie->ie_count == 1 && ie->ie_enable)
       ie->ie_enable(ie);
   }
 }
 
+intr_handler_t *intr_event_add_handler(intr_event_t *ie, ih_filter_t *filter,
+                                       ih_service_t *service, void *arg) {
+  intr_handler_t *ih = kmalloc(M_INTR, sizeof(intr_handler_t), M_ZERO);
+  ih->ih_filter = filter;
+  ih->ih_service = service;
+  ih->ih_argument = arg;
+  ih->ih_name = "?";
```

Comment:
Why the name of a handler isn't passed as an argument?
`INTR_HANDLER_INIT` did support it. I don't see a reason to abandon it.

Reply:
That's true. Please note the PR is still has status _work in progress_ and as such is not finished. I'm thinking on how to restore this functionality.

Reply:
Done.